miden-tx/src/data.rs

```diff
@@ -0,0 +1,8 @@
+use super::{Account, AccountId, BlockHeader, ChainMmr, DataStoreError, NodeIndex, Note};
+
+pub trait DataStore {
+    fn get_block_header(&self, block_num: u32) -> Result<BlockHeader, DataStoreError>;
+    fn get_account(&self, account_id: AccountId) -> Result<Account, DataStoreError>;
+    fn get_note(&self, block_num: u32, note_index: NodeIndex) -> Result<Note, DataStoreError>;
+    fn get_chain_mmr(&self, block_num: u32) -> Result<ChainMmr, DataStoreError>;
```

Comment:
I would actually prefer to have a single method which takes `block_num`, `account_id`, a list of note indexes etc. and returns a single response which contains all required data for executing a transaction. This would help us in the future to simplify communication with the database.

Reply:
I'm guessing we want this unified method in addition to the methods for fetching the data independently for different objects?

Reply:
I was thinking replacing them with the unified method as it would simplify the interface - unless you see value in keeping the individual methods around?

Reply:
I guess it depends if we intend on the `DataStore` also being a provider for RPC requests which may request individual objects?

Reply:
I was thinking of this `DataStore` interface as tailor-built for the `miden-tx` crate (in fact, I'm not sure it should be publicly exported). Internally, implementations of this data store would make requests via RPC to the actual data store which will keep this info in a database.

So, basically, in my mind this `DataStore` would have methods only needed for transaction compilation, execution, and proving.

Reply:
In that case it makes sense to replace them.  I will go ahead and do so.

Reply:
In the future, when we'll start thinking of handling "large accounts" (i.e., accounts with a lot of data), we'd probably need to add more methods here. But for now, I think we'll end up with one or two methods on this data store.